src/pages/index.js

```diff
@@ -82,9 +46,195 @@ const addPlaceModal = new PopupWithForm(
+    })
+    .finally(() => {
+      popupInstance.setLoading(false, defaultText);
+    });
+}
+
+function handleDeleteClick(cardID, cardElement) {
+  deletePlaceModal.setSubmitHandler(() => {
+    function makeRequest() {
+      return api.removePlace(cardID).then(() => {
+        cardElement.remove();
+      });
+    }
+
+    handleSubmit(makeRequest, deletePlaceModal, "Removing...", "Yes");
+  });
+  deletePlaceModal.open();
+}
+
+function getCardElement(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeReact
+  );
+  return card.getView();
+}
+
+/* GET Cards */
+let section;
+api
+  .getInitialCards()
+  .then((items) => {
+    section = new Section(
+      {
+        items: items,
+        renderer: (cardData) => {
+          const cardElement = getCardElement(cardData);
+          return cardElement;
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+  })
+  .catch((err) => console.error(err));
+
+/* GET Profile */
+const userInfo = new UserInfo();
+api
+  .getUserInfo()
+  .then((profileData) => {
+    userInfo.setUserInfo(profileData);
+  })
+  .catch((err) => console.error(err));
+
+/* PATCH Profile Edit Function */
+function handleProfileEditSubmit(profileInputValues) {
+  editProfileModal.setLoading(true);
+  const userData = {
+    name: profileInputValues.title,
+    about: profileInputValues.description,
+  };
+
+  api
+    .editUserInfo(userData.name, userData.about)
+    .then((updatedUserData) => {
+      userInfo.setUserInfo(updatedUserData);
+      editProfileModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info:", err);
+    })
+    .finally(() => {
+      editProfileModal.setLoading(false);
+    });
+}
+
+/* PATCH Profile Avatar Function */
+function handleAvatarSubmit({ url }) {
+  updateAvatarModal.setLoading(true);
+
+  api
+    .updateAvatar(url)
+    .then((res) => {
+      userInfo.updateAvatar(url);
+      updateAvatarModal.reset();
+      updateAvatarModal.close();
+      console.log("Success:", res);
+    })
+    .catch((err) => {
+      console.error("Error updating profile avatar:", err);
+    })
+    .finally(() => {
+      updateAvatarModal.setLoading(true, "Save");
+    });
+}
+
+/* POST Add Place Function */
+function handleNewPlaceSubmit(placeCardData) {
+  function makeRequest() {
+    const cardData = {
+      name: placeCardData.title,
+      link: placeCardData.url,
+    };
+    return api
+      .addNewPlace(cardData.name, cardData.link)
+      .then((newPlaceCard) => {
+        const cardElement = getCardElement({
+          name: newPlaceCard.name,
+          link: newPlaceCard.link,
+          _id: newPlaceCard._id,
+        });
+        section.addItem(cardElement);
+      });
+  }
+
+  handleSubmit(makeRequest, addPlaceModal, "Saving...").then(() => {
+    addPlaceModal.reset();
+    addPlaceModal.setLoading(true, "Save");
+  });
```

Comment:
This `then` will be fullfilled after `catch` which is inside `handleSubmit`. Please, delete this code


You return the button text inside `handleSubmit` in `finally`

`addPlaceModal.reset();` should be in line 195 in `then` (which is before the `catch`)